Fix/apocalypse hail hydra desync

mheguy opened the pull request:
apoc: set data in token rather than passing data, move to Global, change command from some silly string to running it twice. This also gives us a chance to create a save that can be reverted to. Move strings to localised strings

hydra: move data into Global to allow changes during runtime (that don't cause desyncs)

Mostly tested in SP with heavy mode. Would benefit from re-testing as I only tested https://github.com/Refactorio/RedMew/commit/41c88ec682d3f59c5f8dae6f814379b377fb3f12 without heavy mode.

mheguy committed:
Fix desync
apoc: set data in token rather than relying on passed data to be valid
hydra: move data into Global to allow changes during runtime

mheguy committed:
Move to localised strings, command only runs twice now. no special string

grilledham commented:
> hydra: move data into Global to allow changes during runtime (that don't cause desyncs)

This confuses me, wasn't hydra already in global?

plague006 commented:
I'm punching above my weight class trying to describe this, but here goes my attempt:

Hydras were in `global` but the local refs were breaking I think?

Ignoring the apocalypse module entirely, if you simply run `/c global.config.hail_hydra.hydras['behemoth-biter'] = {['behemoth-biter'] = 0.01}` you can get a desync.
When I changed this line <https://github.com/Refactorio/RedMew/pull/756/files#diff-49feeb962373b0ff53ce316e5fbf3877L54> to `local hydra = global.config.hail_hydra.hydras[name]` the desync went away both with the console LUA I pasted as well as the apocalypse module.

While that fixed it, it didn't feel like a strong long-term solution, so I plopped things into `Global` and made some setters.

mheguy committed:
Clean up conditions, change countdown time
Prevent additional saves after apocalypse is active. Give feedback when running command after it is active.
